doc/developers/performance.rst

```diff
@@ -386,10 +386,66 @@ Checkout the official joblib documentation:
 - https://pythonhosted.org/joblib
 
 
-.. _warm-restarts:
+.. _warm-starts:
 
-A sample algorithmic trick: warm restarts for cross validation
+A sample algorithmic trick: warm starts for parameter search
 ==============================================================
 
-TODO: demonstrate the warm restart tricks for cross validation of linear
-regression with Coordinate Descent.
+One of the most common machine learning tasks is parameter search. 
+One approach is to sequentially train an estimator for a list of
+parameters on the same data. Since it is computationally
+expensive, one can speed up by using the ``warm_start``
+parameter of the estimator. Setting ``warm_start=True`` allows to
+reuse the solution of the previous call to fit as initialization.
+For example in the case of linear model with Coordinate Descent,
+warm starting is likely to reduce the number of iteration required
+by the next fit and so leads to faster training.
+:class:`sklearn.linear_model.ElasticNetCV` and other *CV
+(:class:`sklearn.linear_model.LogisticRegressionCV`, ...)
+class already use this technique. The following code shows
+the performance of warm starting during parameter search.
+
+  >>> from sklearn import linear_model
+  >>> from sklearn.datasets.samples_generator import make_regression
+  >>> from sklearn.metrics import mean_squared_error
+  >>> from sklearn.model_selection import KFold
+  >>> from sklearn.grid_search import ParameterGrid
+  >>> import numpy as np
+  >>> import time
+  >>> X, y = make_regression(n_samples=500, n_features=1000, random_state=0)
+  >>> list_params = {"alpha": np.arange(0.01, 1, 0.01)}
```

Comment:
This is okay, but I'd perhaps rather something simpler, with just a fixed (train, test) pair, a smaller dataset, and no `ParameterGrid`, just a list of `alphas`.

Reply:
sorry for the delay. I update it to more simpler example